merlin/models/tf/core/combinators.py

```diff
@@ -453,6 +453,56 @@ def select_by_names(self, names: List[str]) -> Optional[List[Block]]:
                 raise ValueError(f"Given name {name} is not in ParallelBlock {self.name}")
         return blocks
 
+    def select_by_tag(
+        self,
+        tags: Union[str, Tags, List[Union[str, Tags]]],
+    ) -> Optional["ParallelBlock"]:
+        """Select layers of parallel blocks by tags.
+
+        This method will return a ParallelBlock instance with all the branches that
+        have at least one feature that matches any of the tags provided.
+
+        For example, this method can be useful when a ParallelBlock has both item and
+        user features in a two-tower model or DLRM, and we want to select only the item
+        or user features.
+
+        >>> all_inputs = InputBlockV2(schema)  # InputBlock is also a ParallelBlock
+        >>> item_inputs = all_inputs.select_by_tag(Tags.ITEM)
+        ['continuous', 'embeddings']
+        >>> item_inputs.schema["continuous"].column_names
+        ['item_recency']
+        >>> item_inputs.schema["embeddings"].column_names
+        ['item_id', 'item_category', 'item_genres']
+
+        Parameters
+        ----------
+        tags: str or Tags or List[Union[str, Tags]]
+             List of tags that describe which blocks to match
+
+        Returns
+        -------
+        ParallelBlock
+        """
+        if not isinstance(tags, (list, tuple)):
+            tags = [tags]
+
+        selected_branches = {}
+        selected_schemas = Schema()
+
+        for name, branch in self.parallel_dict.items():
+            branch_has_schema = getattr(branch, "has_schema", False)
+            if not branch_has_schema:
+                continue
+            selected_branch = branch.select_by_tag(tags)
```

Comment:
Wouldn’t it be safer to do `hasattr(branch, “select_by_tag”)`?

Reply:
I think a branch being skipped silently because it does not have the method implemented (but has the right tags) is also a concern here, so maybe we should raise an error? (which is what is happening now anyway but the focus would be on a better error message)